File: src/test/java/net/flyingfishflash/loremlist/unit/domain/lrmlistitem/ListItemNotFoundTests.java
package net.flyingfishflash.loremlist.unit.domain.lrmlistitem;

import static org.assertj.core.api.Assertions.assertThat;

import java.util.Set;
import java.util.UUID;
import net.flyingfishflash.loremlist.domain.lrmlistitem.ListItemNotFoundException;
import org.junit.jupiter.api.Test;

class ListItemNotFoundTests {

  @Test
  void idIsNull() {
    ListItemNotFoundException exception = new ListItemNotFoundException();
    assertThat(exception.getResponseMessage()).isEqualTo("ListItem could not be found.");
  }

  @Test
  void idIsNotNull() {
    UUID id = UUID.randomUUID();
    ListItemNotFoundException exception = new ListItemNotFoundException(id);
    assertThat(exception.getResponseMessage()).isEqualTo("ListItem could not be found.");
  }

  @Test
  void customMessage() {
    ListItemNotFoundException exception = new ListItemNotFoundException(Set.of(), "Lorem Ipsum");
    assertThat(exception.getMessage()).isEqualTo("Lorem Ipsum");
    assertThat(exception.getResponseMessage()).isEqualTo("Lorem Ipsum");
  }
}

File: src/test/kotlin/net/flyingfishflash/loremlist/unit/domain/lrmlistitem/ListItemNotFoundTests.kt
package net.flyingfishflash.loremlist.unit.domain.lrmlistitem

import io.kotest.core.spec.style.DescribeSpec
import io.kotest.matchers.shouldBe
import net.flyingfishflash.loremlist.domain.lrmlistitem.ListItemNotFoundException
import java.util.UUID

class ListItemNotFoundTests :
  DescribeSpec({

    describe("ListItemNotFoundException()") {
      it("id is null") {
        val exception = ListItemNotFoundException()
        exception.responseMessage.shouldBe("ListItem could not be found.")
      }

      it("id is not null") {
        val id = UUID.randomUUID()
        val exception = ListItemNotFoundException(id = id)
        exception.responseMessage.shouldBe("ListItem could not be found.")
      }

      it("custom message") {
        val exception = ListItemNotFoundException(message = "Lorem Ipsum")
        exception.message.shouldBe("Lorem Ipsum")
        exception.responseMessage.shouldBe("Lorem Ipsum")
      }
    }
  })
